scripts/docker-integration-tests/aggregator/test.sh

```diff
@@ -0,0 +1,121 @@
+            "zone": "embedded",
+            "weight": 100,
+            "endpoint": "m3aggregator01:6000",
+            "hostname": "m3aggregator01",
+            "port": 6000
+        },
+        {
+            "id": "m3aggregator02:6000",
+            "isolation_group": "availability-zone-b",
+            "zone": "embedded",
+            "weight": 100,
+            "endpoint": "m3aggregator02:6000",
+            "hostname": "m3aggregator02",
+            "port": 6000
+        }
+    ]
+}'
+
+echo "Initializing m3msg topic for m3coordinator ingestion from m3aggregators"
+curl -vvvsSf -X POST localhost:7201/api/v1/topic/init -d '{
+    "numberOfShards": 64
+}'
+
+echo "Initializing m3coordinator topology"
+curl -vvvsSf -X POST localhost:7201/api/v1/services/m3coordinator/placement/init -d '{
+    "instances": [
+        {
+            "id": "m3coordinator01",
+            "zone": "embedded",
+            "endpoint": "m3coordinator01:7507",
+            "hostname": "m3coordinator01",
+            "port": 7507
+        }
+    ]
+}'
+echo "Done initializing m3coordinator topology"
+
+echo "Validating m3coordinator topology"
+[ "$(curl -sSf localhost:7201/api/v1/services/m3coordinator/placement | jq .placement.instances.m3coordinator01.id)" == '"m3coordinator01"' ]
+echo "Done validating topology"
+
+# Do this after placement for m3coordinator is created.
+echo "Adding m3coordinator as a consumer to the aggregator topic"
+curl -vvvsSf -X POST localhost:7201/api/v1/topic -d '{
+  "consumerService": {
+    "serviceId": {
+      "name": "m3coordinator",
+      "environment": "default_env",
+      "zone": "embedded"
+    },
+    "consumptionType": "SHARED",
+    "messageTtlNanos": "600000000000"
+  }
+}' # msgs will be discarded after 600000000000ns = 10mins
+
+echo "Running m3coordinator container"
+echo "> port 7202 is coordinator API"
+echo "> port 7203 is coordinator metrics"
+echo "> port 7204 is coordinator graphite ingest"
+echo "> port 7507 is coordinator m3msg ingest from aggregator ingest"
+docker-compose -f ${COMPOSE_FILE} up -d m3coordinator01
+COORDINATOR_API="localhost:7202"
+
+echo "Running m3aggregator containers"
+docker-compose -f ${COMPOSE_FILE} up -d m3aggregator01
+docker-compose -f ${COMPOSE_FILE} up -d m3aggregator02
+
+echo "Verifying aggregation with remote aggregators"
+
+function read_carbon {
+  target=$1
+  expected_val=$2
+  end=$(date +%s)
+  start=$(($end-1000))
+  RESPONSE=$(curl -sSfg "http://${COORDINATOR_API}/api/v1/graphite/render?target=$target&from=$start&until=$end")
+  test "$(echo "$RESPONSE" | jq ".[0].datapoints | .[][0] | select(. != null)" | tail -n 1)" = "$expected_val"
+  return $?
+}
+
+echo "Sending unaggregated carbon metrics to m3coordinator"
+bash -c 'for i in $(seq 1 100); do echo "foo.bar.baz 42 $(date +%s)" | nc 0.0.0.0 7204; sleep 1; done' &
+
+# Track PID to kill on exit
+METRIC_EMIT_PID="$!"
+
+echo "Read back aggregated metric"
+ATTEMPTS=10 TIMEOUT=1 retry_with_backoff read_carbon foo.bar.* 42
```

Comment:
Why does this work? is the default aggregation policy "last" or something? Can you add a comment

Reply:
Sure can add a comment, the graphite default aggregation policy is average I believe